MekHQ/src/mekhq/gui/view/PersonViewPanel.java

```diff
@@ -345,8 +345,14 @@ private Box drawRibbons() {
                 rowRibbonsBox.setBackground(Color.RED);
             }
             try {
+                int numAwards = person.getAwardController().getNumberOfAwards(award);
```

Comment:
Consider putting this code that is duplicated multiple times into method for readability.  Changes to one of these blocks will likely need the same change to all three similar logic.

Reply:
You're absolutely right. I'll see it done. Thanks for pointing this out. :)